ast/compile.go

```diff
@@ -3680,14 +3700,39 @@ func checkUnusedDeclaredVars(loc *Location, stack *localDeclaredVars, used VarSe
 	return errs
 }
 
-func rewriteSomeDeclStatement(g *localVarGenerator, stack *localDeclaredVars, decl *SomeDecl, errs Errors) Errors {
+func rewriteSomeDeclStatement(g *localVarGenerator, stack *localDeclaredVars, expr *Expr, errs Errors) (*Expr, Errors) {
+	e := expr.Copy()
+	decl := e.Terms.(*SomeDecl)
 	for i := range decl.Symbols {
-		v := decl.Symbols[i].Value.(Var)
-		if _, err := rewriteDeclaredVar(g, stack, v, declaredVar); err != nil {
-			errs = append(errs, NewError(CompileErr, decl.Loc(), err.Error()))
+		switch v := decl.Symbols[i].Value.(type) {
+		case Var:
+			if _, err := rewriteDeclaredVar(g, stack, v, declaredVar); err != nil {
+				return nil, append(errs, NewError(CompileErr, decl.Loc(), err.Error()))
+			}
+		case Call:
+			switch len(v) {
+			case 4: // member3
+				e.Terms = []*Term{
+					RefTerm(VarTerm(Equality.Name)),
+					v[2],
+					RefTerm(v[3], v[1]),
```

Comment:
What does the equality expression look like if we say `some x,y in input.foo`? It looks like we'll construct `eq(v[2], ref(v[3], v[1]))`. Does the ref on the RHS get rewritten later on? E.g.,

```
eq(v[2], ref(v[3], v[1]))
             ^^^^
             this will be a ref
```

So the actual AST would be:

```
eq(v[2], ref(ref(...), v[1]))
```

Reply:
Hrm good point. I'll check that. Refs don't nest like that, indeed. I'll add a test case including an actual non-var, non-constant ref.